test/integration/test_media_api.py

```diff
@@ -0,0 +1,134 @@
+        self.download_file_path = "cat_download.jpeg"
+
+        self.original_file = open(self.media_path + self.media_file, "rb")
+
+    def step1(self) -> None:
+        """Test uploading media
+        """
+        media_id = self.media_id
+        content_type = "image/jpeg"
+        cache_control = "no-cache"
+
+        api_response_with_http_info = self.api_instance.upload_media(
+            account_id=self.account_id,
+            media_id=media_id,
+            body=self.original_file,
+            content_type=content_type,
+            cache_control=cache_control,
+            _return_http_data_only=False
+        )
+
+        self.assertEqual(api_response_with_http_info[1], 204)
+
+        # reopen the media file
+        # the client automatically closes any files passed into request bodies
+        reopened_file = open(self.media_path + self.media_file, "rb")
+
+        # returns void
+        self.api_instance.upload_media(
+            account_id=self.account_id,
+            media_id=media_id,
+            body=reopened_file,
+            content_type=content_type,
+            cache_control=cache_control,
+            _return_http_data_only=False
+        )
+
+    def step2(self) -> None:
+        """Test listing all media on the account
+        """
+        api_response_with_http_info = self.api_instance.list_media(
+            self.account_id, _return_http_data_only=False)
+
+        self.assertEqual(api_response_with_http_info[1], 200)
+
+        api_response = self.api_instance.list_media(self.account_id)
+
+        self.assertIs(type(api_response[0]), Media)
+        pass
+
+    def step3(self) -> None:
+        """Test downloading the media we uploaded in step 1
+        """
+        api_response_with_http_info = self.api_instance.get_media(
+            self.account_id, self.media_id, _return_http_data_only=False)
+
+        self.assertEqual(api_response_with_http_info[1], 200)
+
+        api_response = self.api_instance.get_media(
+            self.account_id, self.media_id, _preload_content=False)
+
+        download_file = open(self.media_path + self.download_file_path, "wb")
+        download_file.write(api_response.data)
+
+        self.assertTrue(filecmp.cmp(self.media_path + self.media_file,
+                        self.media_path + self.download_file_path))
+        download_file.close()
+
+    def step4(self) -> None:
+        """Test deleting the media that was uploaded in step 1
+        """
+        api_response_with_http_info = self.api_instance.delete_media(
+            self.account_id, self.media_id, _return_http_data_only=False)
+
+        self.assertEqual(api_response_with_http_info[1], 204)
+
+        # returns void
+        self.api_instance.delete_media(self.account_id, self.media_id)
+
+    def _steps(self) -> None:
+        for name in dir(self):  # dir() result is implicitly sorted
+            if name.startswith("step"):
```

Comment:
Alternate way to go about finding `functions` you are interested in - override `dir`:
```
   def __dir__(self):
       # this allows naming your functions the way you really want-:)
        return['upload', 'download', 'delete', 'list_media']
```

Reply:
dir() in python returns the list sorted in alphabetic order so i don't think this is possible (with overriding `__dir__` at least - it might be possible another way

Reply:
The above function will return list so the items would be exactly in the specified order. 

Simple way to go about this would be literally calling these functions in the order you desire. It will depict the intention in simple, easy to digest way.